app/src/main/java/com/asfoundation/wallet/ui/balance/TransactionDetailActivity.java

```diff
@@ -312,15 +311,26 @@ private void hideDialog() {
   private void formatValue(String value, String symbol) {
     int smallTitleSize = (int) getResources().getDimension(R.dimen.small_text);
     int color = getResources().getColor(R.color.color_grey_9e);
-
-    amount.setText(BalanceUtils.formatBalance(value, symbol, smallTitleSize, color));
+    String formattedValue = (isSent ? "-" : "+") + value;
+    amount.setText(BalanceUtils.formatBalance(formattedValue, symbol, smallTitleSize, color));
   }
 
-  private String getValue() {
+  private String getValue(String currencySymbol) {
     String rawValue = transaction.getValue();
     if (!rawValue.equals("0")) {
-      rawValue = (isSent ? "-" : "+") + getScaledValue(rawValue);
+      rawValue = getScaledValue(rawValue, currencySymbol);
     }
     return rawValue;
   }
+
+  private WalletCurrency mapToWalletCurrency(String currency) {
```

Comment:
I think this should be a static method inside WalletCurrency. Just like the enum feature valueOf